File: week5/python/682. Baseball Game.py
class Solution:
       def calPoints(self, ops):
        """
        :type ops: List[str]
        :rtype: int
        """
        stack = []
        
        for i in range(len(ops)):
            if ops[i] == "C":
                stack.pop()
            elif ops[i] == "+":
                stackLen = len(stack)
                sumI = stack[stackLen-1] + stack[stackLen-2]
                stack.append(sumI)
            elif ops[i] == "D":
                d = stack[len(stack)-1] * 2
                stack.append(d)
            else:
                stack.append(int(ops[i]))
                
        return sum(stack)

File: week5/Java/682. Baseball Game.java
class Solution {
    public int calPoints(String[] ops) {
        int i=0;
        int[] res = new int[ops.length];
        int ans = 0;
        for(int j=0;j<ops.length;j++,i++){
          String value = ops[j];
          if("C".equals(value)){
            Integer op1 = Integer.valueOf(res[i-1]);
            i-=2;
          }else if("D".equals(value)){
            Integer op1 = Integer.valueOf(res[i-1]);
            res[i] = op1*2;
          }else if("+".equals(value)){
            Integer op1 = Integer.valueOf(res[i-1]);
            Integer op2 = Integer.valueOf(res[i-2]);
            res[i] = op1+op2;
          }else
            res[i] = Integer.valueOf(value); 
        }
      
        while(i--!=0)
          ans+=res[i];
      return ans;
    }
}
